hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreFileManager.java

```diff
@@ -72,7 +70,7 @@ void addCompactionResults(
    * Clears all the files currently in use and returns them.
    * @return The files previously in use.
    */
-  ImmutableCollection<StoreFile> clearFiles();
```

Comment:
StoreFileManager is a private class so this interface change is fine. 

I made the change here for the same reason as other places. We should not be receiving or returning Guava types in our interfaces. This introduces potential compatibility problems. Guava itself evolves and fairly frequently makes breaking changes.